src/stores/ToastStore.ts

```diff
@@ -55,7 +56,7 @@ export default class ToastStore extends EventEmitter {
      *
      * @param {object} newToast The new toast
      */
-    addOrReplaceToast<C extends keyof JSX.IntrinsicElements | JSXElementConstructor<any>>(newToast: IToast<C>) {
+    addOrReplaceToast<C extends ComponentClass>(newToast: IToast<C>) {
```

Comment:
I don't think this type was correct before, and it also does not seem the same as your new store (which is sharing the type at the moment) either...?

This older store seems to take an object of several things, including component class, props, etc. It manually peels off the `component` key, and hands that `createElement`. Your new store expects a single component class directly, which seems quite different, but maybe I am missing something...?

Reply:
it's the right type, just looks really strange. The existing toasts do still take a bunch of properties via the `IToast` class, which is generified to take a `ComponentClass`. The `<C extends ComponentClass>` is more or less a type definition so we can use `C` later in the function's definition.

Reply:
But don't the two toast stores take entirely differently structured objects...?

```
                ToastStore.sharedInstance().addOrReplaceToast({
                    key: 'verifreq_' + request.channel.transactionId,
                    title: request.isSelfVerification ? _t("Self-verification request") : _t("Verification Request"),
                    icon: "verification",
                    props: {request},
                    component: sdk.getComponent("toasts.VerificationRequestToast"),
                    priority: 90,
                });
```

vs. 

```
            const ref = NonUrgentToastStore.instance.addToast(NonUrgentEchoFailureToast);
```

Reply:
`sdk.getComponent` returns the same thing as `import NonUrgentEchoFailureToast` (we're passing the class through, which is super weird)

Reply:
I do not understand. One is an object, with some keys we peel off manually first. The other is a single class we pass straight into `createElement`... How can they possibly be the same?

Reply:
so the `ToastStore` takes an `IToast<C>` (which can be expanded to be `IToast<VerificationRequestToast>` for clarity), where the new one takes just the class, largely for simplicity's sake (we should probably support `props`, but tbh I don't expect this NonUrgentToast stuff to survive).

Reply:
Oh oh oh, okay, I get now, thanks. 😅 😅 😅 Not sure why it took so long.